pages/search/search.jsx

```diff
@@ -51,24 +50,97 @@ export default React.createClass({
       location[`query`] = { keyword: query };
     }
 
+    // note browserHistory.push() triggers component re-render
     browserHistory.push(location);
+  },
+  clearSearch() {
+    this.refs.searchInput.setState({value: ``}, () => {
+      this.updateBrowserHistory();
+    });
+  },
+  fetchData(params = {}, entries = [], apiPageIndex = 1) {
+    if (!params.search) {
+      // if no search keyword was passed reset search page back to its initial state
+      this.setState(this.getInitialState());
+      return;
+    }
+
+    params.page = apiPageIndex;
+
+    Service.entries
+      .get(params)
+      .then((data) => {
+        entries = entries.concat(data.results);
 
+        if (data.next) {
+          // there are more matched entries in the database we need to fetch
+          this.fetchData(params, entries, apiPageIndex+1);
```

Comment:
@mmmavis a lot of paginating APIs use `count` for the total count, with `page` and `per_page` for you to figure out exactly which set in `[0...count]` you're looking at, so I'd recommend we keep it like that.